decam_process.py

```diff
@@ -15,324 +15,378 @@
+            which contains all the compressed defect images.
+
+        This function assumes very particular things about defect ingestion:
+        - They must live in a .tar.gz file in the same location on disk as the other calibs
+        - They will be ingested using ingestCalibs.py run from the calibrepo directory
+        - They will be manually uncompressed and saved in calibrepo/defects/<tarballname>/.
+        - They will be added to the calib registry, but not linked like the flats and biases
+        '''
+        os.chdir(calibrepo)
+        try:
+            os.mkdir('defects')
+        except OSError:
+            # defects directory already exists
+            print('Defects were previously ingested, skipping...')
+        else:
+            print('Ingesting defects...')
+            defectargs = ['../../' + repo, '--calib', '.', '--calibType', 'defect',
+                          '--mode', 'skip', '--validity', '999']
+            defecttarball = defectfiles[0] + '.tar.gz'
+            tarfile.open(os.path.join('../../', defecttarball), 'r').extractall('defects')
+            defectfiles = glob(os.path.join('defects', os.path.basename(defectfiles[0]), '*.fits'))
+            defectargs.extend(defectfiles)
+            defectargumentParser = IngestCalibsArgumentParser(name='ingestCalibs')
+            defectconfig = IngestCalibsConfig()
+            defectconfig.parse.retarget(ingestCalibs.DecamCalibsParseTask)
+            defectIngestTask = IngestCalibsTask(config=defectconfig, name='ingestCalibs')
+            defectParsedCmd = defectargumentParser.parse_args(config=defectconfig, args=defectargs)
+            defectIngestTask.run(defectParsedCmd)
+        finally:
+            os.chdir('../..')
+        return
+
     if not os.path.isdir(calibrepo):
         os.mkdir(calibrepo)
-    print('Ingesting calibration products...')  # NOTE: just biases and flats
-    args = [repo, '--calib', calibrepo, '--mode', 'link', '--validity', '999']
-    args.extend(datafiles)
-    argumentParser = IngestCalibsArgumentParser(name='ingestCalibs')
-    config = IngestCalibsConfig()
-    config.parse.retarget(ingestCalibs.DecamCalibsParseTask)
-    ingestTask = IngestCalibsTask(config=config, name='ingestCalibs')
-    parsedCmd = argumentParser.parse_args(config=config, args=args)
-    try:
-        ingestTask.run(parsedCmd)
-    except sqlite3.IntegrityError as detail:
-        print('~~~ !!! ~~~')
-        print('sqlite3.IntegrityError: ', detail)
-        print('(sqlite3 doesn\'t think all the calibration files are unique)')
-        print('If this isn\'t your first time ingesting these calibration')
-        print('  files, move or delete the existing database and try again.')
-        print('If this is your first time ingesting these calibration files,')
-        print('  make sure you only use image or wtmap MasterCals, not both.')
-        print('~~~ !!! ~~~')
+        flatBiasIngest(repo, calibrepo, calibdatafiles)
```

Comment:
Maybe not for this ticket but I think all of this code would benefit from going into a class, instead of having passing around all the different variables from one function to the next.